tests/testthat/test-trailing_blank_lines_linter.R

```diff
@@ -17,22 +17,39 @@ test_that("returns the correct linting", {
 
   # Construct a test file without terminal newline
   # cf. test-get_source_expressions.R
-  writeLines("lm(y ~ x)", tmp <- tempfile())
-  on.exit(unlink(tmp), add = TRUE)
-  writeBin(
-    # strip the last (two) element(s) (\r\n or \n)
-    head(
-      readBin(tmp, raw(), file.size(tmp)),
-      if (.Platform$OS.type == "windows") -2L else -1L
-    ),
-    tmp2 <- tempfile()
-  )
-  on.exit(unlink(tmp2), add = TRUE)
+  tmp <- withr::local_tempfile()
```

Comment:
Let's set a withr 2.5.0 dependency for the next release and start using `local_tempfile(lines = "lm(y ~ x)")` etc?

Reply:
hmm, maybe let's set that for follow-up. It's not super-appropriate here because we get nice parallelism with `cat()` that I don't think is possible with `lines=`. but we should still do it elsewhere.